glTF/glTF-Structural-Metadata-Samples-Sandcastle.js

```diff
@@ -0,0 +1,229 @@
+// is the index of `FEATURE_ID_n`
+let currentActiveFeatureIdLabel = 0;
+
+// Creates the tileset from the tileset.json in the given subdirectory
+function createTileset(subdirectory) {
+  if (Cesium.defined(currentTileset)) {
+    viewer.scene.primitives.remove(currentTileset);
+    currentTileset = undefined;
+  }
+  // Create the tileset, and move it to a certain position on the globe
+  currentTileset = viewer.scene.primitives.add(
+    new Cesium.Cesium3DTileset({
+      url: `http://localhost:8003/${subdirectory}/tileset.json`,
+      debugShowBoundingVolume: true,
+    })
+  );
+  currentTileset.modelMatrix = Cesium.Transforms.eastNorthUpToFixedFrame(
+    Cesium.Cartesian3.fromDegrees(-75.152325, 39.94704, 0)
+  );
+  const offset = new Cesium.HeadingPitchRange(
+    0,
+    Cesium.Math.toRadians(-22.5),
+    4.0
+  );
+  viewer.zoomTo(currentTileset, offset);
+
+  // Make sure that picking refers to the FEATURE_ID index that
+  // is currently selected in the UI
+  currentTileset.featureIdLabel = currentActiveFeatureIdLabel;
+}
+
+// Create an HTML element that will serve as the
+// tooltip that displays the metadata information
+function createTooltip() {
+  const tooltip = document.createElement("div");
+  viewer.container.appendChild(tooltip);
+  tooltip.style.backgroundColor = "black";
+  tooltip.style.position = "absolute";
+  tooltip.style.left = "0";
+  tooltip.style.top = "0";
+  tooltip.style.padding = "14px";
+  tooltip.style["pointer-events"] = "none";
+  tooltip.style["block-size"] = "fit-content";
+  return tooltip;
+}
+const tooltip = createTooltip();
+
+// Show the given HTML content in the tooltip
+// at the given screen position
+function showTooltip(screenX, screenY, htmlContent) {
+  tooltip.style.display = "block";
+  tooltip.style.left = `${screenX}px`;
+  tooltip.style.top = `${screenY}px`;
+  tooltip.innerHTML = htmlContent;
+}
+
+// Create an HTML string that contains information
+// about the given metadata, under the given title
+function createMetadataHtml(title, metadata) {
+  if (!Cesium.defined(metadata)) {
+    return `(No ${title})<br>`;
+  }
+  const propertyKeys = metadata.getPropertyNames();
+  if (!Cesium.defined(propertyKeys)) {
+    return `(No properties for ${title})<br>`;
+  }
+  let html = `<b>${title}:</b><br>`;
+  for (let i = 0; i < propertyKeys.length; i++) {
+    const propertyKey = propertyKeys[i];
+    const propertyValue = metadata.getProperty(propertyKey);
+    html += `&nbsp;&nbsp;${propertyKey} : ${propertyValue}<br>`;
+  }
+  return html;
+}
+
+// Given an object that was obtained via Scene#pick: If it is
+// a Cesium3DTileFeature, then it is returned.
+// Otherwise, 'undefined' is returned.
+function obtainFeatureMetadata(picked) {
+  if (!Cesium.defined(picked)) {
+    return undefined;
+  }
+  const metadata = picked;
+  const isFeatureMetadata = metadata instanceof Cesium.Cesium3DTileFeature;
+  if (!isFeatureMetadata) {
+    return undefined;
+  }
+  return metadata;
+}
```

Comment:
In these functions, I wouldn't call the variable `metadata`, I would use `feature`. 

a `Cesium3DTileFeature` represents a single feature. it has metadata (accessed via `getProperty()`, but I wouldn't say it _is_ metadata. I just don't want to confuse it for other private API classes like `StructuralMetadata` or `MetadataTable`